engine/src/main/java/io/camunda/zeebe/engine/state/migration/to_8_3/MultiTenancyMigration.java

```diff
@@ -35,5 +51,11 @@ public void runMigration(final MutableProcessingState processingState) {
     migrationState.migrateMessageEventSubscriptionForMultiTenancy();
     migrationState.migrateProcessMessageSubscriptionForMultiTenancy();
     migrationState.migrateJobStateForMultiTenancy();
+
+    // These migrations are not related to MT! However, they perform a migration within a
+    // ColumnFamily. Because of this there is no easy way to detect if they should run, or not. By
+    // making them part of this MigrationTask we can use the 'needToRun' of this task.
+    processingState.getJobState().cleanupTimeoutsWithoutJobs();
+    processingState.getJobState().cleanupBackoffsWithoutJobs();
```

Comment:
❌ In order to make this safe, we need to add an additional condition to the `needsToRun`. Consider the following:
- leader runs multi-tenancy migrations because it has process or decisions deployed and stored in deprecated CF
- it makes a snapshot
- follower on an older version installs the snapshot
- follower applies events without cleaning up job timeout/backoff CFs as it's on older version, but not new process/decision is deployed
- follower upgrades to newer version and becomes leader
- it doesn't run the migration because no new process/decision was deployed, but job timeout/backoff CF are still dirty